scapy/tools/packet_viewer/viewer.py

```diff
@@ -1,17 +1,13 @@
-from typing import List, Union, Tuple, Optional, Dict, Callable
+from typing import List, Union, Tuple, Dict, Callable
 from urwid import AttrMap, MainLoop
 
-from scapy.packet import Raw
 from scapy.supersocket import SuperSocket
+from scapy.compat import bytes_hex
 from scapy.tools.packet_viewer.viewlayer.views.main_view import MainWindow
 
 
 def viewer(
     socket,  # type: SuperSocket
-    columns=None,  # type: Optional[List[Tuple[str, str, str]]]
```

Comment:
Weiß nicht, ob das eine gute Idee ist, `kwargs` erst später "aufzulösen". Weil dann muss man sich für, wenn man `viewer` verwenden möchte, erst selbst durch tiefere Aufruf-Ebenen die Argumente zusammen suchen. Auch fehlen hier jetzt die Typen-Annotations. Finde die expliziten Argumente schöner als implizite. Damit ist es einfacher den Code zu verstehen.
Ist auch bisher klar getrennt gewesen. Explizit -> viewer, implizit -> sniff

Reply:
Hab mir hierzu nochmal gedanken gemacht. Die Pythonic Guidelines sagen klar, explicit vor implicit. 
In diesem Fall hier, leitet jedoch der `viewer` alle Argumente an `MainWindow` weiter. 
Dies begruendet auch, warum mir es intuitiver vorkommt, `**kwargs` zu verwenden. Es handelt sich hier direkt um eine Replikation der Argumente, die nur den Zweck haben, an `MainWindow` weitergeleitet zu werden. (Das replizieren von Argumenten zur Weiterleitung wuerde ich versuchen zu vermeiden, da sonst doppelter Pflegeaufwand für die Interfaces entsteht. Spätestens wenn Dokumentation erstellt wird, erhöht sich der Pflegeaufwand bei kleinen Änderungen drastisch.)
Mit diesem Hintergrund sehe ich noch eine zweite Möglichkeit den Code an dieser Stelle zu verbessern.  Anstatt Argumente für `MainWindow` entgegen zu nehmen, wäre es möglich, das `MainWindow` ausserhalb des `viewers` zusammen zu bauen, und dem `viewer` das konstruierte Objekt als Argument zu uebergeben. 
Was denkst du dazu?

Reply:
Ich hab das jetzt so verstanden, dass du `MainWindow` vom Nutzer aufbauen lassen willst und der Nutzer soll das auch an `viewer` übergeben.
Das macht die Einfachheit des Tools kaputt. Statt `viewer(...)` müsste man sich dann auch um `MainWindow` kümmern und das auch übergeben. Und auf der anderen Seite bringt es für den Anwender keine Vorteile. Außer vielleicht, dass er sich selbst ein anderes `MainWindow` bauen kann, allerdings rechtfertigt ein so seltener Use-Case die zusätzliche Komplexität nicht.
Oder ich hab auch einfach deine Idee nicht richtig verstanden, das kann auch sein.

Reply:
Mir fällt gerade auf, dass `kwargs` ein sehr generische Name ist, der natürlich nur Konvention ist. Wie wärs, wenn man dem einen guten Namen gibt. Wie `**args_for_sniff`. Denke dadurch würde es lesbarer/verständlicher werden und gleichzeitig auch den allgemeinen Charakter verlieren.

Reply:
durch `**args_for_sniff` wuerden die Argumente fuer `MainWindow` und `AsyncSniffer` vermischt werden. Das waere dann noch schlimmer, da der Name `args_for_sniff` impliziert, dass es sich nur um die Argumente des Sniffers handelt

Reply:
wenn ich mir den Code so ansehe, sollte `viewer` eigentlich alles weiterleiten. Nicht mal `socket` wird von `viewer` verwendet. Somit sollte `viewer` dann so aussehen: 
```python
viewer(*args, **kwargs): 
    ...
    main_window = AttrMap(MainWindow(*args, **kwargs), "default")  # type: AttrMap
    ...
```

Reply:
Ich versteh nicht, wie die sich vermischen würden.
Im bisherigen Code (ohne deine Änderungen) werden die `kwargs` nur dem `AsyncSniffer` übergeben, ansonsten nirgends.
`viewer(*args, **kwargs):` finde ich nicht so schön. Weil man sich dann alles zusammen suchen muss und auch keine Typ-Annotations machen kann. Wie du schon gesagt hast, explizit vor implizit. Ich denke `*args` bzw `**kwargs` machen vor allem Sinn, wenn man noch nicht weiß welche Argumente kommen. Wir wissen das hier allerdings sehr gut.

Reply:
Ah, jetzt habe ich es verstanden. Ich denk darüber nach.